client/resolve.c

```diff
@@ -472,3 +457,88 @@ TDNFPrepareSinglePkg(
     }
     goto cleanup;
 }
+
+uint32_t
+TDNFResolveBuildDependencies(
+    PTDNF pTdnf,
+    char **ppszPackageNameSpecs,
+    char **ppszPkgsNotResolved,
+    Queue* queueGoal
+    )
+{
+    uint32_t dwError = 0;
+    int i;
+    PSolvQuery pQuery = NULL;
+    PSolvPackageList pPkgList = NULL;
+    Queue qDeps = {0};
+    const char *pszDep = NULL;
+
+    if(!pTdnf || !pTdnf->pSack || !ppszPackageNameSpecs)
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    if (ppszPackageNameSpecs[0]) {
+        dwError = SolvCreateQuery(pTdnf->pSack, &pQuery);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        dwError = TDNFApplyScopeFilter(pQuery, SCOPE_SOURCE);
```

Comment:
take a look at these entries:
```
0 api.c TDNFListInternal  547 dwError = TDNFApplyScopeFilter(pQuery, nScope);
1 api.c TDNFRepoSync     1109 dwError = TDNFApplyScopeFilter(pQuery, SCOPE_ALL);
2 api.c TDNFRepoQuery    1431 dwError = TDNFApplyScopeFilter(pQuery, nScope);
3 api.c TDNFMark         2315 dwError = TDNFApplyScopeFilter(pQuery, SCOPE_INSTALLED);
```

duplicate code. I think this can be moved to a function of it's own

Reply:
Yes, I have seen that. Another tech debt.